.circleci/src/workflows/discovery.yml

```diff
@@ -0,0 +1,83 @@
+when: << pipeline.parameters.run-discovery-workflow >>
+jobs:
+  - push-docker-image:
+      name: push-discovery-provider
+      context: [GCP, dockerhub]
+      instance-name: circleci-push-discovery-provider-$CIRCLE_BUILD_NUM
+      service: discovery-provider
+  - push-docker-image:
+      name: push-discovery-provider-notifications
+      context: [GCP, dockerhub]
+      instance-name: circleci-push-discovery-provider-notifications-$CIRCLE_BUILD_NUM
+      service: discovery-provider-notifications
+  - push-docker-image:
+      name: push-comms
+      context: [GCP, dockerhub]
+      instance-name: circleci-push-comms-$CIRCLE_BUILD_NUM
+      service: comms
+
+  - test:
+      name: test-discovery-provider
+      context: GCP
+      instance-name: circleci-test-discovery-provider-$CIRCLE_BUILD_NUM
+      service: discovery-provider
+  - test:
```

Comment:
Can multiple jobs have the same name? In client we prefix each job with the filename e.g. `mobile-init`

Reply:
the name is different here right? one is test-dp the other is test-dp-notifications

Reply:
Yeah you're right. I do like the idea of namespacing the jobs though. If two jobs end up with the same name, one gets overwritten I believe. Especially with jobs defined in multiple files, it can be kinda hard to keep track

But not a huge deal, we can leave as is for now!

Reply:
it does, but not if in different workflows. im down to play with this as a follow up if that's good with you?

Reply:
discovery-test-discovery-provider makes me sad